packages/vue-renderer/src/renderer.js

```diff
@@ -418,6 +418,11 @@ export default class VueRenderer {
     // Call ssr:csp hook
     await this.context.nuxt.callHook('vue-renderer:ssr:csp', cspScriptSrcHashes)
 
+    // Add csp meta tags
+    if (this.context.options.render.csp) {
```

Comment:
# Only render if CSP is used

You did it right 👍 

# Feature flag
As I mentioned, a feature flag would be nice. A proper name could be `asMeta` as the CSP string is rendered **as meta** element.

Sth. like:

```suggestion
    if (this.context.options.render.csp && this.context.options.render.csp.asMeta) {
```

# Bonus

Try to enable that one by default in case we use `nuxt generate`. A starting point would be this part:
https://github.com/nuxt/nuxt.js/blob/dev/packages/config/src/options.js#L190-L192

# Bonus Bonus

We need tests for that as well ☺️

Reply:
@manniL Currently the `options.render` object lacks the `asMeta` property right?
> Try to enable that one by default in case we use nuxt generate

I'm not pretty sure on what you intend here :confused: 

Pardon me for all the pain here. I'm not that familiar with the code base and that requires me to ask such questions :smiley:

Reply:
@jamesgeorge007 No worries! Happy that you ask and keep on going ☺️ 	I know that a new large code base can be quite intimidating sometimes 🙈 

> Currently the options.render object lacks the asMeta property right?

That's right. I should be part of the `render.csp` object (if `render.csp` isn't set to false). That's why I made the check above. First check if `render.csp` is truthy (`{} == true`), then check if the object has an `asMeta` entry. In fact, there is not more needed for that change then the change I applied in the suggestion ☺️ 	

> I'm not pretty sure on what you intend here 😕

When using `nuxt generate` you can't use the "header" CSP approach that Nuxt uses by default as no server is running. That's why I suggested to enable that meta element with the CSP string by default when generating the page (and when CSP is enabled ofc).
Hope that makes sense ☺️

Reply:
@manniL As per what I infer what you intend is to place these `if` conditionals as given below just after these lines:-

https://github.com/nuxt/nuxt.js/blob/dev/packages/config/src/options.js#L190-L192

```js
if (render.csp) {
  if (render.csp.hasOwnProperty('asMeta')) {
   // Not exactly sure what should go here 
  }
}
```
> Still I'm not able to connect the dots

Reply:
Am I required to make these workarounds within the `config/src/options.js` file :thinking:

Reply:
@jamesgeorge007 The current state is good!

- [x] Check whether CSP is enabled
- [x] Feature flag (`render.csp.asMeta`)
- [ ] Enable by default (when CSP truthy) on `nuxt generate`
- [ ] Tests
- [ ] Docs

Reply:
For the last "logic" part (enable on generate by default), you have to use `options.js`, yes ☺️